pkg/apis/core/validation/validation.go

```diff
@@ -4184,100 +4186,166 @@ func validateHostIPs(pod *core.Pod) field.ErrorList {
 // tricks.
 // The pod metadata is needed to validate generic ephemeral volumes. It is optional
 // and should be left empty unless the spec is from a real pod object.
-func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+func ValidatePodSpec(pod *core.Pod, podMeta *metav1.ObjectMeta, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
```

Comment:
rather than changing the first arg to *core.Pod, if pod status is *conditionally* needed to do some aspects of validation, making that an optional secondary arg might be clearer (just like how podMeta is an optional secondary arg currently)

Reply:
Fixed